### ISSUE TITLE
Configurations are broken in AppDesigner for .NET Core

### ISSUE THREAD

**Issue by davkean** [dotnet/roslyn-project-system#413]
These pages should be configuration dependent:

![image](https://cloud.githubusercontent.com/assets/1103906/18107049/2432e1b0-6ebb-11e6-87ea-c59b1a1b2c35.png)

Maybe CPS isn't able to detect configurations due to the implicit configurations?

**Comment by RaulPerez1** [comment]
Problem is here:

Public Function GetPropertyPages() As PropertyPageInfo()

Dim ConfigPageGuids As Guid() = GetPageGuids(GetActiveConfigBrowseObject())
Dim CommonPageGuids As Guid() = GetPageGuids(GetProjectBrowseObject())

The code is expecting a couple things:
1) The common pages come before the config pages
2) That the browse object controls which pages are config specific

The project object is coming from the hierarchy (CPS) and has all the pages while the active config one is coming from DTE. Even if the config call was returning any of the pages as long as the project one returns them the code will tag those pages as non config pages.

**Comment by jinujoseph** [comment]
This behaviour you saw after you turned the configuration to  true [here](https://github.com/dotnet/roslyn-project-system/blob/master/src/Microsoft.VisualStudio.ProjectSystem.CSharp.VS/ProjectSystem/VS/Properties/CSharpProjectDesignerPage.cs#L13

**Comment by RaulPerez1** [comment]
Some more data - Debugging through this I'm finding a bunch more issues:
- The asserts on the config specific properties are not specific to the config. A bunch of things are not doing anything at all but not asserting because there is no check on that code path.
- The properties object returned by CPS does not implement VSLangProj.ProjectProperties so _projectPropertiesObject is set to null and then it eventually circles back to returning the raw object anyway (potential perf improvement here).
- For config pages we're getting a ProjectConfig as opposed to a DynamicTypeBrowseObject. The former only has general project info rather than actual properties so even after bypassing the asserts everything is disabled because there aren't descriptors for the properties that drive the controls.

**Comment by Ziflin** [comment]
I posted an issue about some of this on developercommunity.visualstudio.com for VS2017 RC. I'll paste the issues that I've run into here as well since I was directed here.

It seems that it currently impossible to set a Debug or Release-only #define variable/symbol for .Net Core projects in VS 2017 RC. Attempting to set one from the "Project's Properties / Build / Conditional compilation symbols:" entry results in it adding a <DefineConstants> node in the .csproj that is not conditional on the build's configuration. This node now also includes "TRACE;DEBUG;NEWDEFINE" which is very bad as now a Release build will also have TRACE and DEBUG set.

In addition to this, the VS Project Properties dialog does not properly refresh when the Current Build Configuration (from the toolbar drop-down) is changed. This is very confusing as it shows incorrect settings. Closing the dialog and re-opening it will show the proper settings for the newly selected build configuration.
